### PR TITLE
Switcher onboarding

### PR DESCRIPTION
**Pull request by sampettersson**
Updates to the switcher onboarding flow, this PR is so that we can start integrating this into web-onboarding.

- data flow
- some random fixes

### CODE REVIEW

File: src/Components/Actions/ExternalInsuranceProviderAction/AuthStep.tsx [hunk 1]
```diff
@@ -20,35 +22,43 @@ const Title = styled.h4`
...
+    if (!operation?.data?.status) {
+      return;
+    }
 
-    return () => {
-      clearTimeout(timer);
-    };
-  }, []);
+    if (
+      operation?.data?.status != ExternalInsuranceProviderStatus.REQUIRES_AUTH
```
Comment (palmenhq) [inline src/Components/Actions/ExternalInsuranceProviderAction/AuthStep.tsx]: `!==`

File: src/Components/Actions/ExternalInsuranceProviderAction/BackgroundFetchNotification.tsx [hunk 2]
```diff
@@ -92,12 +92,14 @@ export const BackgroundFetchNotification = () => {
             <Body>
               <Title>{currentOperation.provider.name}</Title>
               <Subtitle>
-                {currentOperation.status == BackgroundFetchStatus.ONGOING
+                {currentOperation.data?.status ==
```
Comment (palmenhq) [inline src/Components/Actions/ExternalInsuranceProviderAction/BackgroundFetchNotification.tsx]: `===`

File: src/Components/Actions/ExternalInsuranceProviderAction/BackgroundFetchNotification.tsx [hunk 3]
```diff
@@ -92,12 +92,14 @@ export const BackgroundFetchNotification = () => {
...
+                ExternalInsuranceProviderStatus.FETCHING
                   ? "Vi hämtar din försäkring..."
-                  : "Vi har hämtat..."}
+                  : `Vi hittade din försäkring hos ${currentOperation.provider.name}.`}
               </Subtitle>
             </Body>
-            {currentOperation.status == BackgroundFetchStatus.ONGOING && (
+            {currentOperation.data?.status ==
```
Comment (palmenhq) [inline src/Components/Actions/ExternalInsuranceProviderAction/BackgroundFetchNotification.tsx]: you know the drill :D

File: src/Components/Actions/ExternalInsuranceProviderAction/AuthStep.tsx [hunk 4]
```diff
@@ -20,35 +21,43 @@ const Title = styled.h4`
...
     externalInsuranceProviderAuthScanBankID,
     externalInsuranceProviderAuthOpenBankId
   } = React.useContext(KeywordsContext);
+  const { operation } = React.useContext(DataFetchContext);
 
   React.useEffect(() => {
-    const timer = setTimeout(() => {
-      onDone();
-    }, 10000);
+    if (!operation?.data?.status) {
```
Comment (palmenhq) [inline src/Components/Actions/ExternalInsuranceProviderAction/AuthStep.tsx]: what if status is `CONNECTING`? Then it'll technically be 0 and hence falsy. I think we should check for that explicitly if thats the intention. Or just extend the below check with something like

```
if (
  operation?.data?.status !== ExternalInsuranceProviderStatus.REQUIRES_AUTH
  && operation?.data?.status !== undefined
  && operation?.data?.status !== null
) ...
```

Comment (sampettersson) [reply src/Components/Actions/ExternalInsuranceProviderAction/AuthStep.tsx]: Damn monkey language, backed the enum by strings instead 🤠

**Pull request merged**